pkg/graveler/retention/garbage_collection_manager_test.go

```diff
@@ -0,0 +1,88 @@
+package retention_test
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/block/mem"
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/graveler/retention"
+	"github.com/treeverse/lakefs/pkg/graveler/testutil"
+)
+
+func TestGarbageCollectionManager_GetUncommittedLocation(t *testing.T) {
```

Comment:
Please make this a matrix of tests and add `/` shenanigans. `prefix` is one of (`test_prefix`, `test_prefix/`, `/ test_prefix/`), storage namespace is one of (`mem://test-namespace/my-repo`, `mem://test-namespace/my-repo/`, `mem://test-namespace/`, `mem://test-namespace`), etc.